dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java

```diff
@@ -543,6 +543,38 @@ public Map<String, Object> deleteEdge(User loginUser, long projectCode, long pro
         return result;
     }
 
+    /**
+     * query task relation by projectCode and processDefinitionCode
+     *
+     * @param loginUser loginUser
+     * @param projectCode projectCode
+     * @param processDefinitionCode processDefinitionCode
+     * @return  process task relation lis
+     */
+    @Override
+    public Map<String, Object> queryProcessTaskRelation(User loginUser, long projectCode,
+        long processDefinitionCode) {
+
+        Map<String, Object> result = new HashMap<>();
+
+        ProcessDefinition processDefinition = processDefinitionMapper.queryByCode(processDefinitionCode);
```

Comment:
DS usually checks both interface and resource permissions in service level for security purpose, see:
https://github.com/apache/dolphinscheduler/blob/7586710d49ea86a22798a3f401a875e2f8669dbc/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java#L123
Without permission check,  one user could easily hack other's data.

Reply:
> DS usually checks both interface and resource permissions in service level for security purpose, see:
> 
> https://github.com/apache/dolphinscheduler/blob/7586710d49ea86a22798a3f401a875e2f8669dbc/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java#L123
> 
> 
> Without permission check, one user could easily hack other's data.

OK, please confirm whether my requirements can be integrated. If they can be integrated, I will modify them as soon as possible